### PR TITLE
Prepare for next version

### PR DESCRIPTION
**Pull request by akx**
* Upgrade CI bits and bobs
* Drop support for Python 3.6
* Build & publish wheels from CI (when tagged)

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -59,11 +59,10 @@ jobs:
     env:
       PIP_DISABLE_PIP_VERSION_CHECK: '1'
     steps:
-      - name: 'Set up Python'
```
Comment (ruksi) [inline .github/workflows/ci.yml:62]: the step `name` drops in the change but not that it matters 📛❌

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,3 @@
 [build-system]
-requires = [
-    "setuptools>=42",
-    "wheel"
-]
+requires = ["setuptools>=65"]
```
Comment (ruksi) [inline pyproject.toml:2]: `wheel` is already in the `setuptools` :+1:

File: setup.cfg [hunk 3]
```diff
@@ -2,6 +2,8 @@
 name = minique
 version = attr:minique.__version__
 description = Minimal Redis job runner
+long_description = file:README.md
+long_description_content_type = text/markdown
```
Comment (ruksi) [inline setup.cfg:6]: make that PyPI page sparkling! ✨

File: setup.cfg [hunk 4]
```diff
@@ -20,7 +20,7 @@ include_package_data = True
 
 [options.extras_require]
 sentry =
-    sentry_sdk~=0.19.0
+    sentry_sdk~=1.10
```
Comment (ruksi) [inline setup.cfg:23]: major version bump and no changes needed? amazing 🤩

**Review by ruksi — CHANGES_REQUESTED** [review]
more of a discussion point 📝

moving linting to versioned pre-commit plugins, all good 👍️

I'm not sure dropping dev dependency versions (`pytest` here) is a good idea, or, _I assume_ that it's not an improvement, or what are the upsides? are there any?

at least I don't see them being added to the extras although the requirement files are dropped (in a commit that doesn't mention dropping the testing tool versioning?)

for example, just over the weekend had to manually pin `twine` because the latest `twine` requirements are incompatible with the latest `flake8` requirements (through indirect `importlib-metadata` dependency)

and that was just first one that comes to mind; now CI can more easily break on dev dependency upgrade, all developers will have different versions of testing libraries, etc. etc.

ultimately not a deal breaker but I feels worse than it was :thinking: especially from a potential contributor viewpoint 

example scenario A:

- "what should I install for development if I want to contribute?"
- "check the CI configuration; steps X and Y"
- "which versions?"
- "latest 🤷" (`mypy` version is listed though)

or what do you think?

File: setup.cfg (hunk 4 above)
Comment (akx) [reply setup.cfg:23]: those sentry folks know what they're doing

**Comment by akx** [comment]
> I'm not sure dropping dev dependency versions (`pytest` here) is a good idea, or, _I assume_ that it's not an improvement, or what are the upsides? are there any?

Pytest tends to be very stable, so getting the newest version is usually fine. Either way, I changed things now to loosely pin it to any 6+ version. WDYT?

> * "what should I install for development if I want to contribute?"

Updated the readme a bit around that too 👍

**Comment by ruksi** [comment]
> Pytest tends to be very stable, so getting the newest version is usually fine. Either way, I changed things now to loosely pin it to any 6+ version. WDYT?

it was already acceptable, now it's better :+1: 

`pytest`  will probably be the only dependency anyway (might end up some basic stuff like `pytest-mock` or `freezegun` for more elaborate testing but I doubt it 🤔 )

> Updated the readme a bit around that too +1

awesome, thank you 🙇

**Review by ruksi — APPROVED** [review]

**Pull request merged**